### DISCUSSION TITLE
Building .lib/using the library in projects

### DISCUSSION THREAD

**Discussion by KelticDanor** [libsidplayfp/libsidplayfp#31] [Q&A] [answered]
Hello!

Firstly, I apologise in advance if this question isn't as clear as it could be but hopefully it will make sense.

I've recently been attempting to create a sid plugin for the XMPlay player (un4seen.com), so far the plugin side seems relatively straight forward and the source for foo_sid and been a useful reference in answering some questions already.

My main issue is building the libsidplayfp library for use in the plugin as there is no built .lib files available. I've tried more ways to add it into visual studio 2010/2019 than I could mention and haven't had any luck. At one point even downloading MSYS2 and building it via that, which although it works it produces .a files which I was unable to use.

So long story short, are there any guides/directions you can provide to help me get a static library built in visual studio or otherwise?

Any assistance would be greatly appreciated!

**Comment by drfiemost** [comment]
There used to be a Visual Studio build file back in the old times but the project has changed quite a bit since so I don't think it's of much use today.
I only use MinGW so I can't help, if any user has a build recipe for VS to contribute, I think even a CMakeFile could do, it would be great.

**Comment by drfiemost** [comment]
For what is worth the original build files are still available at sourceforge:
https://sourceforge.net/p/sidplay-residfp/code/HEAD/tree/tags/libsidplayfp/0.3.8/sidplayfp/win/VC/
https://sourceforge.net/p/sidplay-residfp/code/HEAD/tree/tags/libsidplayfp/0.3.8/utils/win/VC/

Too bad they're quite outdated, the source files have been moved around a lot and the two libraries merged together so I doubt they are much useful.

Maybe asking in the Lemon64 forum if someone is building it on VS could be an idea.

**Reply by KelticDanor** [reply]
Hmmm, yes they are very outdated but perhaps they can point me in the right direction, I'll have a look into them regardless.

I'll try contacting kode54 who did the foo_sid plugin as it looks like he has it all set up in VS already which sounds promising. I will definitely take your advice and ask on the Lemon64 forum if I don't have any luck. 

Thank you for your help! I'll leave the question open for a while just in case anyone else has more to add.

**Reply by drfiemost** [reply]
Ok, mentioning @kode54 here in case he's interested in the discussion.

**Comment by emoon** [comment, marked as answer]
I don't know but I use libsidplayfp for my musicplayer and I build it with msvc on windows, clang and mac and gcc on linux and never had any real issue. I don't use cmake, but here is all the config I need to build it https://github.com/HippoPlayer/HippoPlayer/blob/master/units.plugins.lua#L858-L884 and I just have the code here https://github.com/HippoPlayer/HippoPlayer/tree/master/src/plugins/playback/sid/libsidplayfp 

Granted I haven't updated the lib code for a while, but I never had any issues building it really

**Reply by KelticDanor** [reply]
Ah thank you very much @emoon that actually got me looking at the problem a bit differently!

I think I was so focus on getting it working in the one VS scenario I had that I kind of went blind on other anything else, big mistake on my part. I've had a look at your build and it makes a lot more sense to me and it appears I might of been making things more complicated than necessary.

Thank you everyone for your input, it has been invaluable I'll be sure to let you know how things turn out when the plug-in is finished.

**Comment by DomeAlessi** [comment]
@KelticDanor (and for future reference) I had the same problem trying to compile it on Windows and I literally wasted a day trying to figure it out.
After that i found a **solution** that worked like a charm.

Basically I installed this
https://www.msys2.org/#installation
that gets you up and running with all the necessary tools.

You need to follow all the steps:

1 Install the tool.
2 Then run **pacman -Syu** as indicated
3 The also run **pacman -Su** as indicated
4 Then install this **pacman -S --needed base-devel mingw-w64-x86_64-toolchain** 

And then super easy get the latest release of libsidplayfp (in my case was **libsidplayfp-2.2.2.tar.gz**)
Extract it somewhere and with the compiler prompt
5 _[path where you installed MSYS2]_**\mingw64.exe**
do 
6 **cd** _"folder where you extracted the tar.gz file"_

in the prompt do as indicated in the build instructions on .gitHub

7 **./configure
8 make**

You will find the 2 dll file inside
**\src\\.libs\libsidplayfp-6.dll
\src\\.libs\libstilview-0.dll**

additionally on the MSYS website they also have the precompiled package but you still need **pacman** to extract it
https://packages.msys2.org/package/mingw-w64-clang-i686-libsidplayfp?repo=clang32

Hope it helps
